### PR TITLE
Create user/notification rule: allow using existing ones

### PR DESCRIPTION
**Pull request by Padorax**
Similar to https://github.com/heimweh/go-pagerduty/commit/fdaf4fec91903e4232d83876cbf54139d8619b21.

If the new user/notification_rule to be added is duplicate, check and fetch the existing one successfully.
This feature is useful in terraform provider, as it can sync existing user/notification_rule automatically to terraform states.

### CODE REVIEW

File: pagerduty/user.go [hunk 1]
```diff
@@ -299,22 +333,16 @@ func (s *UserService) CreateContactMethod(userID string, contactMethod *ContactM
...
-
-			for _, contact := range resp.ContactMethods {
-				if isSameContactMethod(contact, contactMethod) {
-					return s.GetContactMethod(userID, contact.ID)
-				}
-			}
-
-			return nil, nil, fmt.Errorf("user contact method address is used with different attributes (possibly label)")
-		} else {
+		if e, ok := err.(*Error); !ok || strings.Compare(fmt.Sprintf("%v", e.Errors), "[User Contact method must be unique]") != 0 {
```
Comment (drastawi) [inline pagerduty/user.go:336]: I asked PD if we can hope to return a 409 here in the future as a more reliable error check, but for now this is probably all we can do.

**Review by drastawi — APPROVED** [review]
Looks great. Thanks @Padorax

File: pagerduty/user.go [hunk 2]
```diff
@@ -212,6 +224,28 @@ func (s *UserService) Create(user *User) (*User, *Response, error) {
...
+		return nil, nil, fmt.Errorf("[Email has already been taken] but failed to fetch existing users: %w", lErr)
+	}
+
+	for _, u := range resp.Users {
+		if isSameUser(u, user) {
+			return s.Get(u.ID, &GetUserOptions{})
+		}
+	}
+
+	return nil, nil, fmt.Errorf("[Email has already been taken]")
```
Comment (drastawi) [inline pagerduty/user.go]: @Padorax should we just return error here?
```suggestion
	return nil, nil, error)
```

**Review by drastawi — APPROVED** [review]

File: pagerduty/user.go [hunk 3]
```diff
@@ -383,14 +425,36 @@ func (s *UserService) DeleteContactMethod(userID, contactMethodID string) (*Resp
 	return resp, err
 }
 
+// ListContactMethods lists contact methods for a user.
```
Comment (stmcallister) [inline pagerduty/user.go]: Please match the function name in the comments with the actual function name

**Review by stmcallister — CHANGES_REQUESTED** [review]
@Padorax Thanks for adding this feature! I like the direction you're taking it. However, I'm getting the following error while running the code.

```
pagerduty/user.go:241:19: type error is not an expression
```
Please fix this issue so I can approve the PR. Thanks!!

File: pagerduty/user.go (hunk 3 above)
Comment (Padorax) [reply pagerduty/user.go]: Updated, thanks!

**Comment by Padorax** [comment]
@stmcallister Thanks for the review, all addressed now!

**Comment by Padorax** [comment]
>

**Review by stmcallister — APPROVED** [review]
Looks good! Thanks for fixing that bug!

**Pull request merged**